espresso/models/speech_transformer.py

```diff
@@ -357,6 +365,12 @@ def forward(
                   hidden states of shape `(src_len, batch, embed_dim)`.
                   Only populated if *return_all_hiddens* is True.
         """
+        if self.waveform_inputs:
+            src_tokens = self.logmel_fbank(src_tokens.squeeze())
```

Comment:
To deal with cases that batch_size could be 1, it would better to use src_tokens.squeeze(2) instead.

Reply:
makes sense - will update